packages/flutter/lib/src/painting/decoration.dart

```diff
@@ -198,9 +198,18 @@ abstract class Decoration with Diagnosticable {
 /// happens, the [onChanged] callback will be invoked. To stop this callback
 /// from being called after the painter has been discarded, call [dispose].
 abstract class BoxPainter {
-  /// Abstract const constructor. This constructor enables subclasses to provide
-  /// const constructors so that they can be used in const expressions.
-  const BoxPainter([this.onChanged]);
+  /// Default abstract constructor for box painters.
+  BoxPainter([this.onChanged]) {
```

Comment:
From https://github.com/flutter/flutter/pull/141526#issuecomment-1891184260 it sounds like removing `const` here is a breaking change that would have to follow our breaking change policy (documented in the wiki) if we decide to actually make this change.

FWIW, I don't think we should break this and then break this again when we resolve the TODO below. We should instead resolve the TODO and keep the const. We discussed some ideas for this that wouldn't require a const removal in the past.

Reply:
@goderbauer, thanks for the review and details.
Let's then close this PR and come back to `BoxPainter` in the later stages of leak tracking.